core/codegen/src/ident.rs

```diff
@@ -0,0 +1,128 @@
+//! A variant of Ident with more ergonomic handling of raw identifiers.
+
+use std::cmp::Ordering;
+
+use devise::syn;
+use self::syn::ext::IdentExt as _;
+use crate::proc_macro2::{Ident as Ident2, Span};
+use quote::ToTokens;
+
+#[derive(Clone, Debug)]
+pub struct Ident(Ident2);
+
+impl Ident {
+    pub fn is_raw(&self) -> bool {
+        self.0.to_string().starts_with("r#")
+    }
+
+    pub fn name(&self) -> String {
+        self.0.unraw().to_string()
+    }
+
+    pub fn prepend(&self, string: &str) -> Ident {
+        // TODO: want to use r#{}{}, but that fails (`"r#__rocket_param_enum"` is not a valid identifier)
+        Ident::from(syn::Ident::new(&format!("{}{}", string, self.unraw()), self.span()))
+    }
+
+    pub fn append(&self, string: &str) -> Ident {
+        Ident::from(syn::Ident::new(&format!("{}{}", self, string), self.span()))
+    }
+
+    pub fn span(&self) -> Span {
+        self.0.span()
+    }
+}
+
+impl std::ops::Deref for Ident {
+    type Target = Ident2;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl std::ops::DerefMut for Ident {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+impl From<Ident2> for Ident {
+    fn from(ident: Ident2) -> Ident {
+        Ident(ident)
+    }
+}
+
+impl From<&Ident2> for Ident {
```

Comment:
I find this `impl` kind of odd. Why not `clone()` at the `::from()` site?

Reply:
I added the `impl` per <https://github.com/SergioBenitez/Rocket/pull/894#pullrequestreview-264465565> - I went ahead and used it because it was shorter than `.clone()`, but I don't have a strong opinion on this.